contracts/cw-proposal-multiple/src/contract.rs

```diff
@@ -597,9 +600,53 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
         } => query_reverse_proposals(deps, env, start_before, limit),
         QueryMsg::ProposalHooks {} => to_binary(&PROPOSAL_HOOKS.query_hooks(deps)?),
         QueryMsg::VoteHooks {} => to_binary(&VOTE_HOOKS.query_hooks(deps)?),
+        QueryMsg::FilterListProposals {
+            wallet,
+            status,
+            wallet_vote,
+            start_after,
+            limit,
+        } => {
+            query_filter_list_proposals(deps, env, wallet, status, wallet_vote, start_after, limit)
+        }
     }
 }
 
+fn query_filter_list_proposals(
+    deps: Deps,
+    env: Env,
+    wallet: String,
+    status: Option<Status>,
+    wallet_vote: Option<MultipleChoiceVote>,
+    start_after: Option<u64>,
+    limit: Option<u64>,
+) -> StdResult<Binary> {
+    let wallet: Addr = deps.api.addr_validate(&wallet)?;
+    let min = start_after.map(Bound::exclusive);
+    let limit = limit.unwrap_or(DEFAULT_LIMIT);
+    let props: Vec<ProposalResponse> = PROPOSALS
+        .range(deps.storage, min, None, cosmwasm_std::Order::Ascending)
+        .collect::<StdResult<Vec<(u64, MultipleChoiceProposal)>>>()?
+        .into_iter()
+        .filter_map(|(p_id, prop)| {
+            if status.map_or(true, |st| st == prop.status) {
+                if let Ok(Some(ballot)) = BALLOTS.may_load(deps.storage, (p_id, wallet.clone())) {
+                    if wallet_vote.map_or(true, |v| v == ballot.vote) {
+                        return prop.into_response(&env.block, p_id).ok();
+                    }
+                }
+            }
+            None
+        })
+        .take(limit as usize)
+        .collect();
+    let last_proposal_id = props.last().map_or(0, |res| res.id);
```

Comment:
This should probably be set as the last proposal that we have visited, not the last one we're returning information about. Could we set this based on the IDs we see in `filter_map` above?

For example, currently if we look at proposals 1 through 10 to get proposals 1, 2, and 5 this will return 5. I _think_ we want to return 10 as that is the last one we checked.

Reply:
Thanks, it's very clever!